samza-core/src/test/java/org/apache/samza/processor/TestStreamProcessor.java

```diff
@@ -43,53 +48,85 @@
 import static org.mockito.Mockito.when;
 
 public class TestStreamProcessor {
+  private Map<String, Boolean> processorListenerState;
+  private static final String ON_START = "onStart";
+  private static final String ON_SHUTDOWN = "onShutdown";
+  private static final String ON_FAILURE = "onFailure";
+
+  @Before
+  public void before() {
+    processorListenerState = new HashMap<String, Boolean>() {
+      {
+        put(ON_START, false);
+        put(ON_FAILURE, false);
+        put(ON_SHUTDOWN, false);
+      }
```

Comment:
An alternative here would be to use a Set instead of a Map and assume that membership in the set means that the corresponding event was detected. Clearing out the set is simply a call to .clear(). If you prefer Map then going with an enum and iterating over the values would be less error prone for additional states.